controllers/common.go

```diff
@@ -25,3 +63,105 @@ func ReconcileBatch(l logr.Logger, reconcileFuncs ...ReconcileFunc) (bool, error
 	}
 	return true, nil
 }
+
+func PopulateResticSecret(vsb *datamoverv1alpha1.VolumeSnapshotBackup, vsr *datamoverv1alpha1.VolumeSnapshotRestore) (*corev1.Secret, error) {
+
+	var label, name, namespace string
+	if vsb != nil {
+
+		label = VSBLabel
+		name = vsb.Name
+		namespace = vsb.Spec.ProtectedNamespace
+
+	} else if vsr != nil {
+		label = VSRLabel
+		name = vsr.Name
+		namespace = vsr.Spec.ProtectedNamespace
+	} else {
+		return nil, errors.New("both vsr & vsb reference cannot be empty")
+	}
+	// define Restic secret to be created
+	newResticSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-secret", name),
+			Namespace: namespace,
+			Labels: map[string]string{
+				label: name,
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	return newResticSecret, nil
+}
+
+func BuildVSBResticSecret(givensecret *corev1.Secret, resticsecret *corev1.Secret, pvc *corev1.PersistentVolumeClaim) error {
```

Comment:
Can we not combine the build secret functions for VSB and VSR ?

Reply:
technically we can. It will have a lot of if-else statements and wanted to avoid that.. for VSB side we use VSB CR and PVC..for VSR we need VSR CR..

Reply:
done